### DISCUSSION TITLE
Legacy RS-4 PDA Support?

### DISCUSSION THREAD

**Discussion by clumzy1** [aqualinkd/AqualinkD#288] [General]
Hello, had a few questions I hope are basic and someone can point me in the right direction.

1. Cannot tell exactly what firmware I have but it’s very old- label on manual says RS4 PDA and it is an RS load box but firmware rev on PDA states 1.2…can I use this software with my current setup? (Might try anyway with a rs485 adapter and see what happens)
2. link to upgrade from pda to rs is broken: https://github.com/sfeakes/AqualinkD/wiki/Upgrading-Jandy-Aqualink-PDA-to-RS-panel. —see why this was deleted- please disregard!
3. My Current RS4 setup does not support a VS pump anyway to configure the pump through the rs485 connection? I.e. run a subroutine to control the pump using a different interface than the rs4 connection.

any pointers to the above questions would be appreciated!

thanks in advance

**Comment by sfeakes** [comment]
1) You should be able to, PDA is not the greatest protocol, but one you have a USB adapter, the serial_logger in AqualinkD should tell you everything.  Some early boards had a PDA daughter board over the RS board, it kinda sounds like that’s what you have.
2) sorry, I had to remove that due to jandys lawyers. But a used upgrade board is cheap off eBay.
3) no, not on the same RS bus anyway.  If you run two RS buss then you can use one for the VSP and another for the control panel.  This is because you can only have one master on any given RS 485 bus, and the control panel has to be master and the VSP has to be controlled by the master.

**Reply by clumzy1** [reply]
Thank you so much for the quick response great info!  I want to ask more questions on the upgrade boards since majority of what i am seeing is the new upgrade kits for iaqualink but understand why you can't provide more details and won't ask further!

Once again appreciate it and hopefully i can get this running to test it out!

**Comment by sfeakes** [comment]
The upgrade process that was once on this site, would convert PDA to RS, but it wouldn’t get you VSP support. With enough time I’m sure you can still find enough info on the net to do that conversation.
But a used RS board range from $150 to $700 depending on version, and knowing the version you want is probably the biggest hurdle.  FYI I just bought a used RS8 board, Rev T2 for just under $300 off eBay, and that rev# supports everything any of the newer boards do.  If you want VSP, look for a board that’s rev Q or newer, rev O was when they first introduced VSP support, but it was crap.

being able to decipher all the part# is the problem with buying used.

My first search gave me this link, and it has a lot of information.  And has what you need to know about revisions 
https://www.poolsupplyunlimited.com/Products/Manuals/Sub53_2012329101223.pdf

I just noticed, If you look at the very first column, and compare some of the RS boards against the PDA, that may get you started on the cheap upgrade that was on this site.

**Comment by clumzy1** [comment]
Thank you for the feedback/help. Ended up picking up a RS4 Rev T2g board off ebay - so when that comes in will start putting everything together and hopefully get the VSP and my old V1.29 PDA j-box working again with it.  

Started building a rpi zero 2 W to run out by the pool equipment and with it running bookworm i am having issues running the AqualinkD software.

Following the steps in the wiki - after copying for GIT I receive an error 
"pi@Aqualink-RPI:~/software/AqualinkD $ sudo ~/software/AqualinkD/release/aqualinkd -d -c ~/software/AqualinkD/release/aqualinkd.conf
sudo: unable to execute /home/pi/software/AqualinkD/release/aqualinkd: No such file or directory"

Figured it would just have to be rebuilt so followed instructions below on Wiki and receive a few warnings and an error on the onetouch.c script on line 916 (see below).

![image](https://github.com/sfeakes/AqualinkD/assets/7245802/c0756958-0f6d-4dde-a75f-18ac0e4ac050)

Also tried the "trust everything method" and had similar issues.  Should I drop back to bullseye or earlier debian versions?  THanks!

**Reply by sfeakes** [reply]
To get that error, you must have edited the Makefile and removed RS16 support?
It’s a bug, but if you put the Makefile back to original you will not get that error.

running a compile on bookwork will probably give you another error in the latest release.
See below for fix
https://github.com/sfeakes/AqualinkD/issues/285#issuecomment-2106108273

**Comment by clumzy1** [comment]
Hi!

Thanks for all of the feedback.  Ended up grabbing from ebay a Rev T2 board and configuring the pi 2 zero W and everything is up and running (including the VSP!)

Facing some issues that i think may be wifi connection dropping out and not reconnecting/timing out on reconnecting (only getting -77dB receive strength) and after an undetermined amount of time it drops off and i drop all connections (SSH, webserver and cannot ping).  This I will play with maybe soldering an external antenna connector to the board or mounting outside of the the RS metal enclosure (yeah bad idea to do to begin with with metal attenuation on the signal).

Had a few items I was still playing with that you may save me a few hours of troubleshooting to figure out...
1. Noticed in the aqmanager tab I keep seeing a repeat of the following messages:
```
Notice:PentaDvce: Pentair Pump found at ID 0x60 with RPM 1750 | WATTS 319, but not configured, information ignored!
Notice:PentaDvce: Pentair Pump found at ID 0x60 with RPM 1750 | WATTS 319, but not configured, information ignored!
Notice:PentaDvce: Pentair Pump found at ID 0x60 with RPM 1750 | WATTS 319, but not configured, information ignored!
Warning:iAQ Touch: Got pump message '(Offline)' but can't find pump
Warning:iAQ Touch: Got pump message 'Intelliflo VS 1' but can't find pump at index 1
Warning:iAQ Touch: Got pump message '  Watts: 319' but can't find pump
Warning:iAQ Touch: Got pump message '    RPM: 1750' but can't find pump
Warning:iAQ Touch: Got pump message 'Intelliflo VS 1' but can't find pump at index 1
Warning:iAQ Touch: Got pump message '  Watts: 319' but can't find pump
Warning:iAQ Touch: Got pump message '    RPM: 1750' but can't find pump
Warning:iAQ Touch: Got pump message 'Intelliflo VS 1' but can't find pump at index 1
Notice:PentaDvce: Pentair Pump found at ID 0x60 with RPM 1750 | WATTS 319, but not configured, information ignored!
Notice:PentaDvce: Pentair Pump found at ID 0x60 with RPM 1750 | WATTS 319, but not configured, information ignored!
```

2. Second one has to do with the VSP as well - i can not figure out how to pull up the cron.d configuration on the site so i can set configurations for when to set different VSP speeds and drop them on a normal schedule (i.e. high speed from 6:00-11:00, low speed from 11:00-20:00).  I confirmed in the configuration file i have the "enable_scheduler = yes" and I do have "cron-daemon-common is already the newest version (3.0pl1-162)." so wondering if maybe i'm just not looking at the right URL to pull up a scheduler.

3. Todo - get my old V1.29 J-Box/V3.0 PDA working with the RS system (i think it will be just connecting that up and changing the config file to "pda_sleep_mode = yes" and working through the MQTT documentation now to setup homebridge correctly

Awesome development and thanks for all of the feedback/help with setting it up!.

**Comment by sfeakes** [comment]
For #1 you probably didn’t configure the pump in AqualinkD.conf. Check for this in the button entries. ‘button_01_pumpID=0x60’

For #2 click the time at the top right of AqualinkD web UI. ( in the blue status bar)

**Reply by clumzy1** [reply]
Thanks! I was clicking the blue bar but not on the time hah

**Comment by clumzy1** [comment]
Hi again,

This works amazing! Very impressed by it.  Had some issues with my wifi network dropping out but since have put it in an outdoor weatherproof enclosure and all seems great...next will be installing a watchdog on the wifi signal it if it keeps disconnecting to just reboot the Pi zero 2.  

Looking to configure the Solar_heater function on an RS-4 pool pump.  Currently it has the "spa light" on that relay but going to transition it to be a salt generator so i can control how long to keep the SG powered on for.  As per the spa light, will just splice that into the pool light feed but makes an easy way to see if it works or not now.    

Had just a few follow ups - tried following some discussion posts I found in the discussions topic but it did not end up working for me.

Uncomment the "solar_heater" in /var/www/aqualinkd/config.js and labeled button 8 in the aqualinkd.conf file but it has not changed the gui at all on the hosted site. 


"Spa_Water",
        "Freeze_Protect",
        "CHEM/pH",
        "CHEM/ORP",
        "Solar_Heater",
      ];
"

 /etc/aqualinkd.conf file "button_08_label=Spa Light"

Side question, if I wanted to upgrade from the installed 2.3.4, would you recommend backing up my /etc/aqualinkd.conf and reinstalling everything?

Appreciate all of the answers!

**Comment by sfeakes** [comment]
Due to heavy caching that AqualinkD uses, you will need to clear browser cache every time you make a change to `config.js`, so it could be that. Also on a RS4 only/single panel it would be button 7, on a RS4 Dual/combo it would by button 8, in `aqualinkd.conf`
